### ISSUE TITLE
[Feature Request] (Develop) Attribute `defaultdict` value does not serialize/deserialize correctly

### ISSUE THREAD

**Issue by strikaco** [evennia/evennia#2519]
#### Describe the bug
Storing a `defaultdict` to an Attribute results in the values being lost.

#### To Reproduce
```
>>> from collections import defaultdict
>>> d = defaultdict(list)
>>> c.db.lists = d
>>> c.db.lists
[]
>>> d['name'].append('bob')
>>> c.db.lists = d
>>> c.db.lists
['name']
>>> c.db.lists['name']
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/ubuntu/workspace/evennia/evennia/utils/dbserialize.py", line 224, in __getitem__
    return self._data.__getitem__(key)
TypeError: list indices must be integers or slices, not str
```

#### Expected behavior
Was mainly curious to see if it works.

Store the object correctly or raise an error about it being unserializable.

#### Environment, Evennia version, OS etc
Develop

**Comment by Griatch** [comment]
This is not a bug. https://www.evennia.com/docs/latest/Attributes.html#storing-multiple-objects outlines which iterables can be stored in an Attribute; this does not include defaultdict. This could be a feature request by all means.

**Comment by strikaco** [comment]
> This is not a bug. https://www.evennia.com/docs/latest/Attributes.html#storing-multiple-objects outlines which iterables can be stored in an Attribute; this does not include defaultdict. This could be a feature request by all means.

Fair enough; the issue here is that data *appears* to be stored but is instead silently _lost_. I was surprised to find this was allowed without raising an error.

**Comment by Griatch** [comment]
It should be simple enough to add support for defaultdict serialization in `evennia/utils/dbserialize.py`

**Comment by Griatch** [comment]
Closed in #2577
